src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -0,0 +1,545 @@
+                final DetailAST lastChildOfLand = ancestorOfInstanceOfAst.getLastChild();
+                final DetailAST grandChildOfLand = lastChildOfLand.getLastChild();
+                if (grandChildOfLand == null) {
+                    scopeLineEnd = lastChildOfLand.getLineNo();
+                    scopeColumnEnd = lastChildOfLand.getColumnNo();
+                }
+                else {
+                    scopeLineEnd = grandChildOfLand.getLineNo();
+                    scopeColumnEnd = grandChildOfLand.getColumnNo();
+                }
+            }
+            ancestorOfInstanceOfAst = ancestorOfInstanceOfAst.getParent();
+        }
+
+        variableDesc.setScopeOfPatternVariableIncludingLineAndColumn(scopeLineStart,
+                scopeColumnStart, scopeLineEnd, scopeColumnEnd);
+
+        setScopeOfPatternVariableExcludingLineAndColumn(variableDesc, ancestorOfInstanceOfAst);
+    }
+
+    /**
+     * Set the scope of pattern variable excluding the information
+     * about its line and column number.
+     *
+     * @param variableDesc information about the local variable
+     * @param ancestorOfInstanceOfAst ancestor of DetailAst of type
+     *                                {@link TokenTypes#LITERAL_INSTANCEOF}
+     */
+    private static void setScopeOfPatternVariableExcludingLineAndColumn(
+            VariableDesc variableDesc, DetailAST ancestorOfInstanceOfAst) {
+
+        DetailAST ast = ancestorOfInstanceOfAst;
+        DetailAST grandParent = ast.getParent();
+        boolean fullScopeVariable = ast.getType() == TokenTypes.EXPR
+                && !TokenUtil.isOfType(grandParent.getType(), TokenTypes.ELIST,
+                TokenTypes.LITERAL_DO);
+
+        while (ast.getType() != TokenTypes.EXPR) {
+            ast = ast.getParent();
+        }
+        grandParent = ast.getParent();
+        final DetailAST scopeAst;
+        if (fullScopeVariable) {
+            scopeAst = grandParent.findFirstToken(TokenTypes.SLIST);
+        }
+        else if (grandParent.getType() == TokenTypes.ELIST) {
+            scopeAst = grandParent;
+        }
+        else {
+            scopeAst = ast;
+        }
+        variableDesc.setScopeOfVariable(scopeAst);
+
+    }
+
+    /**
+     * Checks whether the DetailAst of type {@link TokenTypes#IDENT} is
+     * used as left-hand side value. An identifier is being used as a left-hand side
+     * value if it is used as the left operand of an assignment or as the
+     * operand of a stand-alone increment
+     *
+     * @param identAst DetailAst of type {@link TokenTypes#IDENT}
+     * @return True if identAst is used as a left-hand side value otherwise false.
+     */
+    private static boolean isLeftHandSideValue(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        return TokenUtil.isOfType(parent.getType(), INCREMENT_AND_DECREMENT_KINDS)
+                || parent.getType() == TokenTypes.ASSIGN
```

Comment:
`parent.getType() == TokenTypes.ASSIGN` you can add assign token to existing array of INCREMENT_AND_DECREMENT tokens (and rename it)
it will be easier to read

Reply:
The condition is different-
```
return TokenUtil.isOfType(parent.getType(), INCREMENT_AND_DECREMENT_TOKENS)
                || (parent.getType() == TokenTypes.ASSIGN
                && identAst != parent.getLastChild());
```